docarray/computation/tensorflow_backend.py

```diff
@@ -0,0 +1,280 @@
+import typing
+from typing import Callable, List, Optional, Tuple
+
+import numpy as np
+import tensorflow as tf  # type: ignore
+import tensorflow._api.v2.experimental.numpy as tnp  # type: ignore
+
+from docarray.computation import AbstractComputationalBackend
+from docarray.computation.abstract_numpy_based_backend import AbstractNumpyBasedBackend
+from docarray.typing import TensorFlowTensor
+
+
+def _unsqueeze_if_single_axis(*matrices: tf.Tensor) -> List[tf.Tensor]:
+    """
+    Unsqueezes tensors that only have one axis, at dim 0.
+    This ensures that all outputs can be treated as matrices, not vectors.
+
+    :param matrices: Matrices to be unsqueezed
+    :return: List of the input matrices,
+        where single axis matrices are unsqueezed at dim 0.
+    """
+    unsqueezed = []
+    for m in matrices:
+        if len(m.shape) == 1:
+            unsqueezed.append(tf.expand_dims(m, axis=0))
+        else:
+            unsqueezed.append(m)
+    return unsqueezed
+
+
+def _unsqueeze_if_scalar(t: tf.Tensor) -> tf.Tensor:
+    """
+    Unsqueezes tensor of a scalar, from shape () to shape (1,).
+
+    :param t: tensor to unsqueeze.
+    :return: unsqueezed tf.Tensor
+    """
+    if len(t.shape) == 0:  # avoid scalar output
+        t = tf.expand_dims(t, 0)
+    return t
+
+
+def norm_left(t: tf.Tensor) -> TensorFlowTensor:
+    return TensorFlowTensor(tensor=t)
+
+
+def norm_right(t: TensorFlowTensor) -> tf.Tensor:
+    return t.tensor
+
+
+class TensorFlowCompBackend(AbstractNumpyBasedBackend[TensorFlowTensor]):
```

Comment:
general comment about the class, to me it looks good I like that we rely on tnp as much as we can.

Maybe norm_left, norm_right should be named differenlty

norm_left could be : cast_output
norm_right could be: get_tensor

this will make the code clearer

maybe @JohannesMessner as a better idea for the name.